### PR TITLE
refactor: Improving Code level as per PR comments for condition checks, var declarations, regex comments

### PR DESCRIPTION
**Pull request by imnavanath**
### Description
- Direct `Astra_Builder_Helper::$is_header_footer_builder_active` conditions avoided and used with strict bool checks `true === Astra_Builder_Helper::$is_header_footer_builder_active`
- Removed possible else part of conditions where variables are declared
- Set those variables before `if` condition and if condition met then only updated in conditional part
- Updated valid variable names & removed all `temp` variable strings from vars
- Added regex comments in overall JS scripts

### Types of changes
- Code level changes
- Non-breaking

### How has this been tested?
- With builder customizer (for JS errors if trigger any)
- With all components used in Builder layout

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Review by vrundakansara — APPROVED** [review]

File: inc/core/class-astra-enqueue-scripts.php [hunk 1]
```diff
@@ -126,7 +126,20 @@ public function admin_body_class( $classes ) {
...
 		public static function theme_assets() {
 
-			if ( Astra_Builder_Helper::$is_header_footer_builder_active ) {
+			$default_assets = array(
+
+				// handle => location ( in /assets/js/ ) ( without .js ext).
+				'js'  => array(
+					'astra-theme-js' => 'style',
+				),
+
```
Comment (premanshup) [inline inc/core/class-astra-enqueue-scripts.php:135]:
```suggestion

```

File: inc/core/class-astra-enqueue-scripts.php [hunk 2]
```diff
@@ -126,7 +126,20 @@ public function admin_body_class( $classes ) {
 		 */
 		public static function theme_assets() {
 
-			if ( Astra_Builder_Helper::$is_header_footer_builder_active ) {
+			$default_assets = array(
+
```
Comment (premanshup) [inline inc/core/class-astra-enqueue-scripts.php:130]:
```suggestion

```

File: inc/customizer/extend-custom-controls/src/builder-layout/builder-component.js [hunk 3]
```diff
@@ -120,71 +120,71 @@ const BuilderComponent = props => {
 
 		AstraBuilderCustomizerData.component_limit = parseInt(AstraBuilderCustomizerData.component_limit);
 
-		let tmp_choices = (AstraBuilderCustomizerData && AstraBuilderCustomizerData.choices && AstraBuilderCustomizerData.choices[controlParams.group] ? AstraBuilderCustomizerData.choices[controlParams.group] : []);
+		let customizerBuilderchoices = (AstraBuilderCustomizerData && AstraBuilderCustomizerData.choices && AstraBuilderCustomizerData.choices[controlParams.group] ? AstraBuilderCustomizerData.choices[controlParams.group] : []);
```
Comment (premanshup) [inline inc/customizer/extend-custom-controls/src/builder-layout/builder-component.js:123]: Can be `const`

File: inc/customizer/extend-custom-controls/src/builder-layout/drop-component.js [hunk 4]
```diff
@@ -9,35 +9,35 @@ const DropComponent = props => {
 	const location = props.zone.replace(props.row + '_', '');
 	let currentList = typeof props.items != "undefined" && props.items != null && props.items.length != null && props.items.length > 0 ? props.items : [];
 	let choices = props.choices,
-		tmpChoices = Object.keys(choices),
+		filterChoices = Object.keys(choices),
 		theItems = [];
 	{
-		let tmpCurrentList = [...currentList];
+		let sortList = [...currentList];
```
Comment (premanshup) [inline inc/customizer/extend-custom-controls/src/builder-layout/drop-component.js:15]: Can be `const`

File: inc/customizer/extend-custom-controls/src/builder-layout/item-component.js [hunk 5]
```diff
@@ -40,11 +40,11 @@ const ItemComponent = props => {
 				existing_component_count = existing_component_count - 1;
 				component_track[ forceRemoveSection.builder + '-' + forceRemoveSection.type ] = existing_component_count;
```
Comment (premanshup) [inline inc/customizer/extend-custom-controls/src/builder-layout/item-component.js:41]: should be `camelCase`

File: inc/customizer/extend-custom-controls/src/customizer.js [hunk 6]
```diff
@@ -196,7 +196,7 @@
 
 			if (false !== is_cloning_index) {
 				let clone_from_id = id;
-				clone_from_id = clone_from_id.replace(/[0-9]+/g, is_cloning_index);
+				clone_from_id = clone_from_id.replace(/[0-9]+/g, is_cloning_index); // Replace random numeric with valid clone index.
```
Comment (premanshup) [inline inc/customizer/extend-custom-controls/src/customizer.js:199]: should be `camelCase`

**Review by premanshup — CHANGES_REQUESTED** [review]

File: inc/core/class-astra-enqueue-scripts.php (hunk 1 above)
Comment (imnavanath) [reply inc/core/class-astra-enqueue-scripts.php]: Done 👍

File: inc/core/class-astra-enqueue-scripts.php (hunk 2 above)
Comment (imnavanath) [reply inc/core/class-astra-enqueue-scripts.php]: Done 👍

File: inc/customizer/extend-custom-controls/src/builder-layout/builder-component.js (hunk 3 above)
Comment (imnavanath) [reply inc/customizer/extend-custom-controls/src/builder-layout/builder-component.js]: Done 👍

File: inc/customizer/extend-custom-controls/src/builder-layout/drop-component.js (hunk 4 above)
Comment (imnavanath) [reply inc/customizer/extend-custom-controls/src/builder-layout/drop-component.js]: Done 👍

File: inc/customizer/extend-custom-controls/src/builder-layout/item-component.js (hunk 5 above)
Comment (imnavanath) [reply inc/customizer/extend-custom-controls/src/builder-layout/item-component.js]: Done 👍

File: inc/customizer/extend-custom-controls/src/customizer.js (hunk 6 above)
Comment (imnavanath) [reply inc/customizer/extend-custom-controls/src/customizer.js]: Done 👍

File: inc/customizer/configurations/layout/class-astra-site-identity-configs.php [hunk 7]
```diff
@@ -270,6 +263,18 @@ public function register_configuration( $configurations, $wp_customize ) {
 					),
 				),
 
+				/**
+				 * Option: Divider
+				 */
+				array(
```
Comment (rohitp-bsf) [inline inc/customizer/configurations/layout/class-astra-site-identity-configs.php:269]: This config needs to be removed! if merged later on.

**Review by rohitp-bsf — CHANGES_REQUESTED** [review]

File: inc/customizer/configurations/layout/class-astra-site-identity-configs.php (hunk 7 above)
Comment (imnavanath) [reply inc/customizer/configurations/layout/class-astra-site-identity-configs.php:269]: @rohitp-bsf I think this position is valid one for divider - https://a.cl.ly/xQunJEkA
Why this should be removed?

Comment (rohitp-bsf) [reply inc/customizer/configurations/layout/class-astra-site-identity-configs.php:269]: In another task, we have moved dividers into CSS-based configs. Nevermind will merge and update it in final PR.

**Review by rohitp-bsf — APPROVED** [review]

**Review by premanshup — APPROVED** [review]

**Pull request merged**